documentation/specs/Modal.md

```diff
@@ -0,0 +1,191 @@
+# `Modal` Component Specification
+
+## Overview
+
+A `Modal` is a page overlay that displays information and blocks interaction with the page until an action is taken or the `Modal` is dismissed.
+
+## Design
+
+`Modal` will use `useDialog` and `useModalOverlay` from `react-aria` to provide the technical foundation for an accessible dialog.
+
+The component design was inspired both by Shopify's Modal component and Twilio's Modal component.
+
+`Modal` will be a controlled component deferring the state management to the consumer.
+
+`Modal` will be a compound component consistenting of `Modal`, `Modal.Header`, `Modal.Body`, and `Modal.Footer`.
+
+### API
+
+```ts
+type HeaderElementType = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+type ModalProps = {
+  /**
+   * Whether the modal is open or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * Callback when the modal is closed.
+   */
+  onClose: () => void;
+
+  /**
+   * Whether or not the modal can be dismissed.
+   *
+   * @default true
+   */
+  isDismissable?: boolean;
```

Comment:
i don't love having a boolean prop that defaults to `true`. i rather the API require the boolean to be active to make it read better in the tree. however i couldn't think of a clean name that wasn't a negation like `isNotDismissable` which read less clean IMO